build/cmake/setup.cmake

```
@@ -646,15 +646,8 @@ if(wxUSE_XLOCALE)
     set(CMAKE_EXTRA_INCLUDE_FILES)
 endif()
 
-# Check size and availability of various types
-set(SYSTYPES
-    pid_t size_t
-    wchar_t int long short
-    gid_t uid_t
-    )
-if(NOT MSVC)
-    list(APPEND SYSTYPES mode_t off_t)
-endif()
+# Check sizes of various types
+set(SYSTYPES size_t wchar_t int long short)
```

```suggestion
set(SYSTYPES wchar_t int long short)
```
`size_t` is still needed here? Seems it will also be defined in sys/types.h on Unix system.

```
/* Define to `unsigned' if <sys/types.h> doesn't define.  */
#cmakedefine size_t unsigned
```

---

We need `SIZEOF_SIZE_T`, so we do need it here AFAICS (this is also the reason we have `int` etc here, these types are always present, of course, but we want to know their sizes).

---

Thanks for your clarification. @vadz